tests/integration/backup/test_v3_backup.py

```diff
@@ -0,0 +1,306 @@
+    metadata_path = backup_path / f"{topic_name}.metadata"
+    with metadata_path.open("rb") as buffer:
+        metadata = read_metadata(buffer)
+    (data_file,) = metadata_path.parent.glob("*.data")
+
+    # Make sure topic doesn't exist beforehand.
+    try:
+        admin_client.delete_topics([topic_name])
+    except UnknownTopicOrPartitionError:
+        print("No previously existing topic.")
+    else:
+        print("Deleted topic from previous run.")
+
+    # Execute backup restoration.
+    subprocess.run(
+        [
+            "karapace_schema_backup",
+            "restore",
+            "--config",
+            str(config_file),
+            "--topic",
+            topic_name,
+            "--location",
+            str(metadata_path),
+        ],
+        capture_output=True,
+        check=True,
+    )
+
+    # Execute backup creation.
+    backup_start_time = datetime.datetime.now(datetime.timezone.utc)
+    subprocess.run(
+        [
+            "karapace_schema_backup",
+            "get",
+            "--use-format-v3",
+            "--config",
+            str(config_file),
+            "--topic",
+            topic_name,
+            "--location",
+            str(tmp_path),
+        ],
+        capture_output=True,
+        check=True,
+    )
+    backup_end_time = datetime.datetime.now(datetime.timezone.utc)
+
+    # Verify only expected files exist in backup target directory.
+    assert sorted(path.name for path in tmp_path.iterdir()) == [
+        f"{topic_name}.metadata",
+        f"{topic_name}:0.data",
+    ]
+
+    # Parse metadata from file.
+    (new_metadata_path,) = tmp_path.glob("*.metadata")
+    with new_metadata_path.open("rb") as buffer:
+        new_metadata = read_metadata(buffer)
+
+    # Verify start and end timestamps are within expected range.
+    assert backup_start_time < new_metadata.started_at
+    assert new_metadata.started_at < new_metadata.finished_at
+    assert new_metadata.finished_at < backup_end_time
+
+    # Verify new version matches current version of Karapace.
+    assert new_metadata.tool_version == __version__
+
+    # Verify all fields other than timings and version match exactly.
+    for field in fields(Metadata):
+        if field.name in {"started_at", "finished_at", "tool_version"}:
+            continue
+        assert getattr(metadata, field.name) == getattr(new_metadata, field.name)
+
+    # Verify data files are identical.
+    (new_data_file,) = new_metadata_path.parent.glob("*.data")
+    assert data_file.read_bytes() == new_data_file.read_bytes()
+
+
+def test_can_inspect_metadata() -> None:
+    topic_name = "2db42756"
+    backup_path = Path(__file__).parent.parent.resolve() / "test_data" / "backup_v3_single_partition"
+    metadata_path = backup_path / f"{topic_name}.metadata"
+
+    # Execute inspect command, and expect exit status 0. We don't check the output
+    # further as it varies across platforms.
```

Comment:
This is due to forced coloring, should use same solution as in https://github.com/aiven/karapace/pull/613